### DISCUSSION TITLE
Sharp and Promise.all in route handler

### DISCUSSION THREAD

**Discussion by galaczi** [adonisjs/core#3674] [Help] [answered]
I am trying to create multiple sizes of a post featured image. Originally I broke it up to smaller functions but for this issue it's simpler this way.

Inside a route handler:

```
const sizes = [[400, 400], [600, 600], [800, 800], [1200, 1200], [1600, 1600]]

await Promise.all(sizes.map(size => sharp(featuredimage.tmpPath)
  .resize(...size)
  .toFile('public/uploads/' + post.slug + size[0] + 'x' + size[1] + '.webp')))

response.redirect().toRoute('posts.index')

```

It creates the files correctly but for some reason the redirect wont work. Instead I am redirected back to the edit page (like if I used ``response.redirect().back()``. If I take out Promise.all and resize to only one size, the redirect happens as expected.

I am probably missing something about Promises, am I?

**Comment by thetutlage** [comment, marked as answer]
> It creates the files correctly but for some reason the redirect wont work

You mean no redirect happens at all? Do you see any error on the page or something?

**Reply by galaczi** [reply]
A page refresh happens on submit, however I am redirected back to the same (edit) page, instead of the specified ``posts.index``. There is no error at all and if I add a console.log before the ``response.redirect`` the console.log gets triggered.

**Reply by thetutlage** [reply]
Seems weird. Can you create reproduce this inside a fresh project and share the repo with me?

**Reply by galaczi** [reply]
I was not able to reproduce the issue in a clean project so now I am trying to figure out what I am doing differently in my project. I will report back once I figure it out.

**Reply by galaczi** [reply]
I had a lot of moving parts but finally I was able to reproduce it by including Webpack Encore in a barebones project: https://github.com/galaczi/sharp-adonis-scripts

Seems to be frontend related. When I comment out ``@entryPointScripts('app')`` in [show.edge](https://github.com/galaczi/sharp-adonis-scripts/blob/main/resources/views/show.edge) the redirect happens. Otherwise it doesn't and there are cancelled network requests. It happens in Chrome but not in Firefox.

**Reply by thetutlage** [reply]
It works fine for me on latest version of Firefox!


https://user-images.githubusercontent.com/1706381/163755980-b6bb2902-773b-4284-8ef9-4002ad49cb57.mov



Also you mentioned that the frontend related code is having some issues. However, the repo you shared does not have a single line of frontend JavaScript. Recommend you to debug it further and share a reproducible example

**Reply by galaczi** [reply]
First of all thanks for looking into it! It works fine on Firefox, the problem is with Chrome based browsers. I created a short video: https://youtu.be/hDUWlRV8ik0
It's a stock Chromium browser with no extensions. It doesn't make much sense to me either. I have spent a lot of time turning off auth, middlewares, etc. until I discovered these network errors.

**Reply by thetutlage** [reply]
Reading this [thread](https://stackoverflow.com/a/13459106) made me realize it can be because of the Webpack dev server reloading because the public folder has changed. 

And that was it. If you write files inside a different folder, maybe `tmp/uploads` (infact this where the user uploaded files should go), then issue will not occur. Also, if you turn off webpack dev server `node ace serve --watch --no-assets`, then also the issue will not occur

**Reply by galaczi** [reply]
That makes sense. Thanks again!
